common/src/main/scala/org/broadinstitute/monster/etl/MsgTransformations.scala

```diff
@@ -44,17 +44,61 @@ object MsgTransformations {
   }
 
   /**
-    * Aggregate the values of a list of fields into a single array field
+    * Remove the key-value pairs of a message for a given field.
+    *
+    * Keys not included in the removal are left in the message as-is.
+    *
+    * If a "remove" key given in fields is not present in the message,
+    * the transformation continues (as opposed to throwing an exception).
+    */
+  def removeFields(fields: Set[String])(msg: Msg): Msg = {
+    val toRet = upack.copy(msg)
+    val underlying = toRet.obj
+    fields.foreach { field =>
+      underlying.remove(Str(field))
+    }
+    toRet
+  }
+
+  /**
+    * Extract the key-value pairs of a message for a given field.
+    *
+    * Keys not included in the fields input are left out.
+    *
+    * If none of the fields to extract are present in the message,
+    * throw an exception.
+    */
+  def extractFields(fields: Set[String])(msg: Msg): Msg = {
+    val toRet = Obj()
+    val underlying = toRet.obj
+    fields.foreach { field =>
+      if (msg.obj.keySet.contains(Str(field))) {
```

Comment:
This `contains` isn't needed. `msg.obj.get(Str(field))` will return `None` if the key isn't present, and the chained `foreach` call will no-op.